client/lib/login/index.js

```diff
@@ -78,6 +78,14 @@ export function getSignupUrl( currentQuery, currentRoute, oauth2Client, locale,
 			includes( get( currentQuery, 'redirect_to' ), '/jetpack/connect/authorize' ) &&
 			includes( get( currentQuery, 'redirect_to' ), '_wp_nonce' )
 		) {
+			// If the current query has plugin_name param, but redirect_to doesn't, add it to the redirect_to
+			const pluginName = get( currentQuery, 'plugin_name' );
+			const urlObj = new URL( currentQuery.redirect_to );
```

Comment:
Perhaps, we can add a try-catch block here to handle the case where the redirect_to is not a valid URL. Of `current Query.redirect_to` is not a valid URL, it will throw an error.

Reply:
@chihsuan good suggestion! Updated it in [Try catch for invalid redirect_to value](https://github.com/Automattic/wp-calypso/pull/95504/commits/c580ced6f6ffcedcb1a59284f6fca5d7e81bc37a)